### PR TITLE
Package with Halcyon

### PR DESCRIPTION
**Pull request by mietek**
- Constrain `utf8-string` to < 1 for _mueval_ ([`hiplambda.cabal`](https://github.com/purefn/hiplambda/pull/4/files#diff-8f413579b3d71ad33de5010aa3fc40ea))
- Constrain `cabal-install` to 1.22.4.0 for GHC 7.10.1 ([`.halcyon/cabal-version`](https://github.com/purefn/hiplambda/pull/4/files#diff-769c30dee475df05ec2b5bc8038a9856))
- Declare version constraints for all Cabal package dependencies ([`.halcyon/constraints`](https://github.com/purefn/hiplambda/pull/4/files#diff-31d23e4c587602195ce9ac67169bd13e))
- Declare `happy-1.19.5` as a build-time dependency ([`.halcyon/sandbox-extra-apps`](https://github.com/purefn/hiplambda/pull/4/files#diff-3239948c6fe2c52ad610d2bcc6534436))
- Declare version constraints for the dependencies of `happy-1.19.5` ([`.halcyon/sandbox-extra-apps-constraints/happy-1.19.5.constraints`](https://github.com/purefn/hiplambda/pull/4/files#diff-e1a658c7cd63d6f21639303302f0a651))
- Install `hoogle-index-0.4.2` as a run-time dependency using a pre-install hook ([`.halcyon/pre-install-hook`](https://github.com/purefn/hiplambda/pull/4/files#diff-4d96efe4faa86e193cd7551d6445ca7f), [`.halcyon/hoogle-index-0.4.2`](https://github.com/purefn/hiplambda/pull/4/files#diff-7b7006b3fb8f8fc47ea387848e416b0a))
- Add missing line endings

### CODE REVIEW

**Comment by mietek** [comment]
Looks like Travis CI needs to be set up to use Halcyon. See [“Building Haskell projects with Halcyon”](https://robots.thoughtbot.com/building-haskell-projects-with-halcyon) for an example.

**Comment by purefn** [comment]
I hadn't considered using Halcyon on Travis CI. Thanks!

File: .halcyon/pre-install-hook [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	local magic_dir
+	magic_dir="${source_dir}/.halcyon/hoogle-index-0.4.2"
+
+	local -a opts_a
+	opts_a=()
+	opts_a+=( --constraints="${magic_dir}/constraints" )
+	opts_a+=( --sandbox-extra-apps="${magic_dir}/sandbox-extra-apps" )
+	opts_a+=( --sandbox-extra-apps-constraints="${magic_dir}/sandbox-extra-apps-constraints" )
+
+	halcyon install "${opts_a[@]}" 'hoogle-index-0.4.2' || return 1
```
Comment (purefn) [inline .halcyon/pre-install-hook:19]: This installs hoogle-index, but it doesn't actually do the indexing, correct?

File: .halcyon/cabal-version [hunk 2]
```diff
@@ -0,0 +1 @@
+1.22.4.0
```
Comment (purefn) [inline .halcyon/cabal-version:1]: Can the GHC version be put into a file? Right now I have it in the config setting, which is less than ideal.

File: .halcyon/pre-install-hook (hunk 1 above)
Comment (mietek) [inline .halcyon/pre-install-hook:19]: That’s right. I haven’t used `hoogle-index` myself, so I wasn’t sure what to do with it.

File: .halcyon/cabal-version (hunk 2 above)
Comment (mietek) [inline .halcyon/cabal-version:1]: Currently, the GHC version is automatically inferred from the version constraint of the `base` package, in your [`constraints`](https://github.com/purefn/hiplambda/blob/master/.halcyon/constraints#L19) file:

```
base-4.8.0.0
```

This is described in the [“Make a change”](https://halcyon.sh/tutorial/#make-a-change) section of the tutorial.  Therefore, your config setting is redundant, and you should be able to get rid of it.

This is likely to change in the future (https://github.com/mietek/halcyon/issues/52#issuecomment-79626776). A `.halcyon/ghc-version` magic file will be accepted then, and you will see the name of the magic file listed in the documentation for [`HALCYON_GHC_VERSION`](https://halcyon.sh/reference/#halcyon_ghc_version).

**Pull request merged**